tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,213 @@
+use crate::Args;
+use buildsys_config::DockerArchitecture;
+use clap::Parser;
+use log::{info, trace};
+use pubsys_config::InfraConfig;
+use regex::Regex;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::fs::read_dir;
+use std::path::{Path, PathBuf};
+use tokio::process::Command;
+
+/// Takes a local kit built using buildsys and publishes it
+/// to a vendor specified in Infra.toml
+#[derive(Debug, Parser)]
+pub(crate) struct PublishKitArgs {
+    /// Location of the local kit
+    #[arg(long)]
+    kit_path: PathBuf,
+    /// Vendor to publish kit to
+    #[arg(long)]
+    vendor: String,
+}
+
+struct OciArchive {
+    filename: String,
+    arch: String,
+    version: String,
+}
+
+macro_rules! docker {
+    ($arg: expr) => {{
+        let result = Command::new("docker")
+            .args($arg)
+            .output()
+            .await
+            .context(error::IoSnafu)?;
+        ensure!(
+            result.status.success(),
+            error::PublishFailSnafu {
+                reason: String::from_utf8_lossy(&result.stderr)
+            }
+        );
+        result.stdout
+    }};
+}
+
+pub(crate) async fn run(args: &Args, publish_kit_args: &PublishKitArgs) -> Result<()> {
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let infra_config = InfraConfig::from_path_or_lock(&args.infra_config_path, false)
+        .context(error::ConfigSnafu)?;
+    trace!("Parsed infra config: {:?}", infra_config);
+    ensure!(infra_config.vendor.is_some(), error::NoVendorsSnafu);
+
+    // Fetch the vendor container registry uri
+    let vendor = infra_config
+        .vendor
+        .as_ref()
+        .unwrap()
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    let vendor_registry_uri = vendor.registry.clone();
+    info!(
```

Comment:
Feels a little overeager to announce finding something in a file.

```suggestion
    debug!(
```